### PR TITLE
new derived table: search_queries_daily

### PR DESCRIPTION
**Pull request by rebecca-burwei**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

### CODE REVIEW

**Comment by jklukas** [comment]
I've refactored this and pushed the changes. Please take a look, @rebecca-burwei.

File: sql/moz-fx-data-shared-prod/search_terms_derived/search_queries_daily_v1/query.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+SELECT
+  DATE(submission_timestamp) AS submission_date,
+  search_query,
+  COUNT(*) AS impressions,
+  COUNTIF(is_clicked) AS clicks,
```
[the remark this answers is not in the document]
Comment (rebecca-burwei) [reply sql/moz-fx-data-shared-prod/search_terms_derived/search_queries_daily_v1/query.sql:5]: lgtm

File: bigquery_etl/dryrun.py [hunk 2]
```diff
@@ -113,7 +113,7 @@
     "sql/moz-fx-data-shared-prod/mozilla_vpn_derived/protected_v1/init.sql",
     "sql/moz-fx-data-shared-prod/mozilla_vpn_derived/add_device_events_v1/init.sql",
     "sql/moz-fx-data-shared-prod/mozilla_vpn_external/devices_v1/init.sql",
-    "sql/moz-fx-data-shared-prod/search_terms/suggest_impression_sanitized/view.sql",
+    *glob.glob("sql/moz-fx-data-shared-prod/search_terms*/**/*.sql", recursive=True),
```
Comment (jklukas) [inline bigquery_etl/dryrun.py:116]: We won't be able to do dry runs on any of these search terms datasets due to the dry run service not having access to the tables we're hitting.

**Comment by rebecca-burwei** [comment]
@jklukas We have `search_terms_daily` in some places and `search_queries_daily` in others. Should we make them all `search_terms_daily`?

**Comment by jklukas** [comment]
> We have search_terms_daily in some places and search_queries_daily in others. Should we make them all search_terms_daily

For the DAG, I wanted `search_terms_daily` to align with the overall dataset name `search_terms`. I agree that it's confusing to have a slightly different name for the table; yet the field within that table is `search_query`. Perhaps the table should be `sanitized_query_counts_daily`? Or `top_search_queries_daily`?

File: sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_aggregated/search_queries_daily_v1/init.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+OPTIONS
+  (require_partition_filter = TRUE)
+AS
+SELECT
+  CAST(NULL AS DATE) AS submission_date,
+  CAST(NULL AS STRING) AS search_query,
+  CAST(NULL AS INT64) AS impressions,
+  CAST(NULL AS INT64) AS clicks,
+  CAST(NULL AS INT64) AS client_days,
+WHERE
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_aggregated/search_queries_daily_v1/init.sql:14]: @whd reports this clause I added causes:

> Query without FROM clause cannot have a WHERE clause

File: sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_aggregated/search_queries_daily_v1/init.sql [hunk 4]
```diff
@@ -0,0 +1,15 @@
+CREATE TABLE IF NOT EXISTS
+  `moz-fx-data-shared-prod.search_terms_derived.search_queries_daily_v1`
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_aggregated/search_queries_daily_v1/init.sql:2]: @rebecca-burwei and I spoke about naming just now and agreed on `aggregated_search_terms_daily` here.

**Comment by whd** [comment]
I've run the `init.sql` manually for now:
`This statement created a new table named moz-fx-data-shared-prod:search_terms_derived.aggregated_search_terms_daily_v1. `

**Review by rebecca-burwei — COMMENTED** [review]
looks good to me

**Review by jklukas — APPROVED** [review]

**Pull request merged**